src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2233,22 +2233,21 @@ internal bool ExpressionRequiresFixing(BoundExpression expr, out Symbol accessed
                         }
                     case BoundKind.PointerElementAccess:
                         {
-                            // When using pointers in unsafe code, it is generally left for the user to make sure memory is fixed/pinned correctly.
-                            // Except for fixed fields, we want to produce appropriate diagnostics if field is not fixed before access.
+                            // C# 7.3:
+                            // a variable resulting from a... pointer_element_access of the form P[E] [is fixed] if P
+                            // is not a fixed size buffer expression, or if the expression is a fixed size buffer
+                            // member_access of the form E.I and E is a fixed variable
                             BoundExpression underlyingExpr = ((BoundPointerElementAccess)expr).Expression;
-                            if (underlyingExpr.Kind == BoundKind.FieldAccess)
+                            if (underlyingExpr is BoundFieldAccess fieldAccess && fieldAccess.FieldSymbol.IsFixed)
                             {
```

Comment:
should we keep the `.Kind` check for perf reasons? no need to do an unnecessary cast here otherwise.

Reply:
I did a comparison a while ago and found that if you end up doing the cast, in certain circumstances it may be faster to use the pattern. It's slower if the pattern check fails, but that's a question of frequency of success vs. failure. Overall, my results in the perf testing were that this wasn't terribly likely to be a problem, unless you're doing it in one of the hottest loops in the compiler.